src/libcore/slice.rs

```diff
@@ -643,6 +643,17 @@ macro_rules! slice_offset {
     }};
 }
 
+macro_rules! slice_sub_offset {
```

Comment:
Could you add a comment about why one can't just use `slice_offset!`/`slice_add_offset` with `-1`?